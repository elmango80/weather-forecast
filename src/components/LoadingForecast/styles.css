.euiLoadingContent__singleLine:last-child {
    margin-bottom: 0;
}

.forecast__card--loading .euiCard__title .euiLoadingContent {
    margin: 0.75rem 0px;
}

.forecast__card--loading .forecast__card-content .euiLoadingContent {
    width: 10rem;
}

.forecast__card--loading .forecast__card-content:last-child {
    margin-top: 0.5rem;
}
