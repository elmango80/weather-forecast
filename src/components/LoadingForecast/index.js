import React from "react";

import {
  EuiButtonIcon,
  EuiCard,
  EuiFlexGroup,
  EuiFlexItem,
  EuiLoadingContent,
} from "@elastic/eui";

import "./styles.css";
import "styles/weather-icons.css";

export default function LoadingForecast() {
  return (
    <EuiCard
      className="forecast__card forecast__card--loading"
      betaBadgeLabel="cargando"
      textAlign="left"
      title={
        <EuiFlexGroup
          gutterSize="l"
          alignItems="stretch"
          justifyContent="spaceBetween"
          responsive={false}
        >
          <EuiFlexItem>
            <EuiLoadingContent lines={1} />
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiButtonIcon
              className="card-title__button-icon"
              iconType="starEmpty"
              iconSize="xl"
              aria-label="Add to favorites"
              disabled
            />
          </EuiFlexItem>
        </EuiFlexGroup>
      }
      description={<></>}
    >
      <EuiFlexGroup
        className="forecast__card-content"
        alignItems="center"
        justifyContent="spaceBetween"
        responsive={false}
      >
        <EuiFlexItem grow={false}>
          <EuiLoadingContent lines={3} />
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiLoadingContent lines={2} />
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiFlexGroup
        className="forecast__card-content"
        alignItems="center"
        justifyContent="spaceBetween"
        responsive={false}
      >
        <EuiFlexItem grow={false}>
          <EuiLoadingContent lines={3} />
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiLoadingContent lines={3} />
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiCard>
  );
}
