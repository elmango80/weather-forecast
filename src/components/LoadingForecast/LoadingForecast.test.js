import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import LoadingForecast from "./index";

describe("Test <LoadingForecast />", () => {
  test("renders correctly", () => {
    const wrapper = shallow(<LoadingForecast />);

    expect(wrapper).toMatchSnapshot();
  });
});
