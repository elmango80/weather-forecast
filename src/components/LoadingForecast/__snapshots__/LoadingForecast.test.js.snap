// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <LoadingForecast /> renders correctly 1`] = `
<Fragment>
  <EuiCard
    betaBadgeLabel="cargando"
    className="forecast__card forecast__card--loading"
    description={
      <React.Fragment>
        Cargando
      </React.Fragment>
    }
    textAlign="left"
    title={
      <EuiFlexGroup
        alignItems="stretch"
        gutterSize="l"
        justifyContent="spaceBetween"
        responsive={false}
      >
        <EuiFlexItem>
          <EuiLoadingContent
            lines={1}
          />
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          <EuiButtonIcon
            aria-label="Add to favorites"
            className="card-title__button-icon"
            disabled={true}
            iconSize="xl"
            iconType="starEmpty"
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    }
  >
    <EuiFlexGroup
      alignItems="center"
      className="forecast__card-content"
      justifyContent="spaceBetween"
      responsive={false}
    >
      <EuiFlexItem
        grow={false}
      >
        <EuiLoadingContent
          lines={3}
        />
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <EuiLoadingContent
          lines={2}
        />
      </EuiFlexItem>
    </EuiFlexGroup>
    <EuiFlexGroup
      alignItems="center"
      className="forecast__card-content"
      justifyContent="spaceBetween"
      responsive={false}
    >
      <EuiFlexItem
        grow={false}
      >
        <EuiLoadingContent
          lines={3}
        />
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <EuiLoadingContent
          lines={3}
        />
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiCard>
</Fragment>
`;
